### PR TITLE
Add ugv_nav4d and dependencies

### PR DESCRIPTION
**Pull request by haider8645**
Hello guys,

added ugv_nav4d. Please check if everything is covered in source.yml and the .autobuild files.

Best,
Haider

### CODE REVIEW

File: source.yml [hunk 1]
```diff
@@ -3,4 +3,18 @@ name: ric
 imports:
     - github: rock-core/package_set
 
+######################
```
Comment (chhtz) [inline source.yml:13]: Here a `version_control:` is missing (otherwise, autoproj thinks the following entries are part of the `imports:` list

Comment (haider8645) [reply source.yml:13]: Thanks! added it

**Comment by chhtz** [comment]
I added some further necessary dependencies.
I assume with Ubuntu20.04 it builds as is (did not check), for Ubuntu22/24 one needs additional qt4-qt5 overrides (which we currently have in our internal package_set)

**Comment by chhtz** [comment]
@pierrewillenbrockdfki Can you merge the latest changes from https://github.com/rock-core/rock-package_set to your qt4-qt5 branch here: https://github.com/pierrewillenbrockdfki/rock_core_rock_package_set?

**Comment by pierrewillenbrockdfki** [comment]
> @pierrewillenbrockdfki Can you merge the latest changes from https://github.com/rock-core/rock-package_set to your qt4-qt5 branch here: https://github.com/pierrewillenbrockdfki/rock_core_rock_package_set?

Done.

**Comment by chhtz** [comment]
With some more changes, this makes a minimal buildconf using this manifest build now on Ubuntu 24:

```
package_sets:
    - github: haider8645/ric-package_set
      branch: add_ugv_nav4d
layout:
  - dfki.ric
  - rock.cpp
  - rock.core
ignored_packages:
 - gui/vizkit
 - gui/qgv
 - gui/rock_replay
```

I did not check Ubuntu 20/22.

`gui/qgv` and `gui/rock_replay` are hopefully fixable, but that should not delay merging this.

(For Ubuntu 24, one needs to use a branch of autoproj. Instructions are here: https://github.com/rock-core/autoproj/issues/408 in the comments)

**Review by chhtz — APPROVED** [review]

**Pull request merged**